repository/src/lib.rs

```diff
@@ -54,16 +54,30 @@ impl<T: RawRepository> DistributedRepository<T> {
 
     /// Initializes the genesis repository from the genesis working tree.
     pub async fn genesis(&mut self) -> Result<(), Error> {
-        unimplemented!()
+        //let genesis_info = self.get_reserved_state().await?.genesis_info;
+        Ok(())
     }
     /// Returns the block header from the `main` branch.
     pub async fn get_last_finalized_block_header(&self) -> Result<BlockHeader, Error> {
-        unimplemented!()
+        let commit_hash = self
+            .raw
+            .locate_branch(&FINALIZED_BRANCH_NAME.into())
+            .await?;
+        let semantic_commit = self.raw.read_semantic_commit(&commit_hash).await?;
+        let block_header: BlockHeader = serde_json::from_str(&semantic_commit.body)?;
+        Ok(block_header)
     }
 
-    /// Returns the reserved state from the `main` branch.
+    /// Returns the reserved state from the 'finalized' branch.
```

Comment:
```suggestion
    /// Returns the reserved state from the `finalized` branch.
```

Reply:
ditto

Reply:
btw fix the implementation of this function since `read_reserved_state` is added